### PR TITLE
fixes #7

### LINKED ISSUE #7: Conflict with Sprout Forms plugin
When the Locale Sync plugin is enabled, and Sprout Forms is enabled, Locale Sync will cause an error when trying to  create a new Sprout Form and change the default form handle.

```
2016/12/15 14:37:14 [error] [system.db.CDbCommand] CDbCommand::fetch() failed: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'klipsch_staging.craft_sproutformscontent_form1' doesn't exist. The SQL statement executed was: SELECT *
FROM `craft_sproutformscontent_form1`
WHERE (`elementId`=:pnlimW2ymc) AND (`locale`=:psNN739cdI).
2016/12/15 14:37:14 [error] [exception.CDbException] CDbException: CDbCommand failed to execute the SQL statement: SQLSTATE[42S02]: Base table or view not found: 1146 Table 'klipsch_staging.craft_sproutformscontent_form1' doesn't exist in /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/framework/db/CDbCommand.php:543
Stack trace:
#0 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/framework/db/CDbCommand.php(415): CDbCommand->queryInternal('fetch', Array, Array)
#1 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/services/ContentService.php(65): CDbCommand->queryRow()
#2 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/models/BaseElementModel.php(815): Craft\ContentService->getContent(Object(Craft\SproutForms_FormModel))
#3 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/framework/base/CComponent.php(111): Craft\BaseElementModel->getContent()
#4 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/models/BaseModel.php(87): CComponent->__get('content')
#5 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/models/BaseElementModel.php(141): Craft\BaseModel->__get('content')
#6 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/plugins/localesync/services/LocaleSyncService.php(121): Craft\BaseElementModel->__get('content')
#7 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/plugins/localesync/services/LocaleSyncService.php(93): Craft\LocaleSyncService->updateElement(Object(Craft\SproutForms_FormModel), 'title')
#8 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/plugins/localesync/LocaleSyncPlugin.php(28): Craft\LocaleSyncService->syncElementContent(Object(Craft\Event), NULL)
#9 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/framework/base/CComponent.php(567): Craft\LocaleSyncPlugin->Craft\{closure}(Object(Craft\Event))
#10 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/services/ElementsService.php(2359): CComponent->raiseEvent('onbeforesaveele...', Object(Craft\Event))
#11 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/services/ElementsService.php(1380): Craft\ElementsService->onBeforeSaveElement(Object(Craft\Event))
#12 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/plugins/sproutforms/services/SproutForms_FormsService.php(172): Craft\ElementsService->saveElement(Object(Craft\SproutForms_FormModel))
#13 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/plugins/sproutforms/controllers/SproutForms_FormsController.php(106): Craft\SproutForms_FormsService->saveForm(Object(Craft\SproutForms_FormModel))
#14 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/framework/web/actions/CInlineAction.php(49): Craft\SproutForms_FormsController->actionSaveForm()
#15 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/framework/web/CController.php(308): CInlineAction->runWithParams(Array)
#16 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/framework/web/CController.php(286): CController->runAction(Object(CInlineAction))
#17 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/framework/web/CController.php(265): CController->runActionWithFilters(Object(CInlineAction), Array)
#18 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/framework/web/CWebApplication.php(282): CController->run('saveForm')
#19 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/etc/web/WebApp.php(817): CWebApplication->runController('sproutForms/for...')
#20 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/etc/web/WebApp.php(287): Craft\WebApp->_processActionRequest()
#21 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/framework/base/CApplication.php(185): Craft\WebApp->processRequest()
#22 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/craft/app/index.php(62): CApplication->run()
#23 /mnt/klipschweb-data-prod1/www/klipsch-staging/app/public_staged/index.php(19): require_once('/mnt/klipschweb...')
#24 {main}
REQUEST_URI=/admin/sproutforms/forms/edit/46798
HTTP_REFERER=http://staging.klipsch.com/admin/sproutforms/forms/edit/46798
```

I was able to get around this with a null check in the Craft\LocaleSyncService->syncElementContent method. Will post a PR soon.

### PR DESCRIPTION
**Pull request by superwick**

### CODE REVIEW

File: services/LocaleSyncService.php [hunk 1]
```diff
@@ -48,59 +48,61 @@ public function getLocaleInputOptions($locales = null, $exclude = [])
 
 	public function syncElementContent(Event $event, $elementSettings)
 	{
-		$pluginSettings = craft()->plugins->getPlugin('localeSync')->getSettings();
-		$this->_element = $event->params['element'];
-		$this->_elementSettings = $elementSettings;
+		if ($elementSettings !== null) {
```
Comment (timkelty) [inline services/LocaleSyncService.php:51]: This diff go really messy with indentations and line offsets.

Is the only real change there wrapping the everything in `if ($elementSettings !== null) {..}`?
If so, I'd rather just bail out up top with:

```
if ($elementSettings === null) {
  return;
}
```

**Review by timkelty — COMMENTED** [review]
This diff go really messy with indentations and line offsets.

Is the only real change there wrapping the everything in `if ($elementSettings !== null) {..}`?
If so, I'd rather just bail out up top with:
```
if ($elementSettings === null) {
  return;
}
```

File: services/LocaleSyncService.php (hunk 1 above)
Comment (superwick) [reply services/LocaleSyncService.php:51]: Yes, that's how I got around it. Your solution is prettier. ;)

**Comment by timkelty** [comment]
Interesting...I'm doing the opposite check here: https://github.com/timkelty/craft-localesync/blob/master/services/LocaleSyncService.php#L55-L58

I wonder if your PR makes breaks syncing with defaults when saving through the HUD?

**Pull request closed without merging**